### ISSUE TITLE
Mumble Positional Audio Support

### ISSUE THREAD

**Issue by moxvallix** [Anuken/Mindustry-Suggestions#2773]
### Describe the content or mechanics you are proposing.

Mumble, the FOSS voice chat has a feature that allows for in game positional audio. Basically this allows people's voices to sound like their position in game eg. they sound quieter as they get further away etc.

However, for Mumble's positional audio to work, the game needs to support it, by exposing player co-ordinates to Mumble.

I would love for Mindustry to support positional audio.

### Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.

This is a great feature when playing online with friends, as it allows for a little more immersion in the game. I think Mindustry could really benefit from supporting positional audio, as being able to hear only those nearby you is great for communicating with others working on the same area as you, without voice clashing with others, that are, for example, up top defending.

### Before making this issue, check the boxes below to confirm that you have acknowledged them.

- [X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
- [X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
- [X] I have read [README.md](https://github.com/Anuken/Mindustry-Suggestions/blob/master/README.md) to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by itcannotbe** [comment]
You do realize that the host needs to be running the server for FOSS

**Comment by moxvallix** [comment]
> You do realize that the host needs to be running the server for FOSS

You can run your own Mumble server, it's called Murmur, and I run my own for my friends. Mumble is completely Free / Open source (their license is a bit weird though, link [here](https://github.com/mumble-voip/mumble/blob/master/LICENSE).

**Comment by itcannotbe** [comment]
Can't this done with mods? Cause you probably need another client anyways for voice.

**Comment by moxvallix** [comment]
> Can't this done with mods?

A mod could do it too, yes, but it's always nice when a game supports it vanilla. 
> Cause you probably need another client anyways for voice.

Voice is handled by mumble, the game only needs to tell mumble the player's position in game.

**Comment by itcannotbe** [comment]
You already need an external client to handle voice but you want a part of it in vanilla?

**Comment by moxvallix** [comment]
@itcannotbe I don't think you understand what Positional Audio support is. Please read https://www.mumble.info/documentation/user/positional-audio/. Basically, the game just tells the voice chat where players are in game, rather than the game providing voice chat.

**Comment by abrasmag3** [comment]
I think mods would be a good option, as different voice chats might have different needs for this kind of support. The mod could probably be lightweight, so it wouldn't be too much of a problem

**Comment by abrasmag3** [comment]
Some ideas for how to handle required variables


Position and rotation:
position and rotation of the currently controlled unit


Camera postion and rotation:
Same as above, but no rotation


"Context":
Server and team (e.g Sharded)


"Identity":
UUID most likely, maybe including the type of controlled unit


All just ideas

**Comment by Anuken** [comment]
I don't use Mumble, and I'm not particularly interested in learning a C(++?) API and creating JNI bindings for an application that I'm not familiar with. Create a mod.
